File: back-end/src/main/java/ma/infosat/gestion/adherent/entities/Adherent.java
package ma.infosat.gestion.adherent.entities;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonSetter;

@Entity
public class Adherent implements Serializable{
	/**
	 * 	 */
	private static final long serialVersionUID = 1L;
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id_adherent;
	private String nom_adherent;
	private String prenom_adherent;
	private String cin;
	private LocalDate dateDeNaissance;
	private String lieuDeNaissance;
	private LocalDate dateAdhesion;
	private String numeroVote;
	private String photo;
	private String motPasse;
	private String sexe;
	private String telephone;
	private String email;
	private String role;
	@OneToMany(mappedBy="adherent",cascade=CascadeType.ALL)
	private Set<Adresse> adresses=new HashSet<Adresse>();
	
	@ManyToOne(cascade=CascadeType.MERGE)
	@JoinColumn(name="communaute_id")
	private Communaute communaute;
	
	@ManyToOne(cascade=CascadeType.MERGE)
	@JoinColumn(name="bureauVote_id")
	private BureauVote bureauVote;
	
	@ManyToOne(cascade=CascadeType.MERGE)
	@JoinColumn(name="id_service")
	private Service service;
	
	@OneToMany(mappedBy="fk.adherent",cascade=CascadeType.ALL)
	private Set<Equipe> equipes=new HashSet<Equipe>();
	
	@ManyToMany(mappedBy="adherents")
	private Set<TypeAdherent> typeAdherents =new HashSet<TypeAdherent>();
	@ManyToOne(cascade=CascadeType.MERGE)
	@JoinColumn(name="prefession_id")
	private Profession profession;
	
	@OneToMany(mappedBy="fk.adherent")
	private Set<LigneFonction> lignesFonction =new HashSet<LigneFonction>();
	public Adherent(String nom_adherent, String prenom_adherent, String cIN, LocalDate dateDeNaissance,
			String lieuDeNaissance,String numeroVote, String photo,String email,String telephone,String sexe,String role) {
		super();
		this.nom_adherent = nom_adherent;
		this.prenom_adherent = prenom_adherent;
		this.cin = cIN;
		this.dateDeNaissance = dateDeNaissance;
		this.lieuDeNaissance = lieuDeNaissance;
		this.dateAdhesion =LocalDate.now();
		this.numeroVote = numeroVote;
		this.email=email;
		this.telephone=telephone;
		this.photo = photo;
		this.sexe=sexe;
		this.role=role;
	}
	
	public Adherent() {
		this.dateAdhesion=LocalDate.now();
	}

	public String getNom_adherent() {
		return nom_adherent;
	}

	public void setNom_adherent(String nom_adherent) {
		this.nom_adherent = nom_adherent;
	}

	public String getPrenom_adherent() {
		return prenom_adherent;
	}

	public void setPrenom_adherent(String prenom_adherent) {
		this.prenom_adherent = prenom_adherent;
	}

	public String getCin() {
		return this.cin;
	}

	public void setCin(String cin) {
		this.cin = cin;
	}

	public LocalDate getDateDeNaissance() {
		return dateDeNaissance;
	}

	public void setDateDeNaissance(LocalDate dateDeNaissance) {
		this.dateDeNaissance = dateDeNaissance;
	}

	public String getLieuDeNaissance() {
		return lieuDeNaissance;
	}

	public void setLieuDeNaissance(String lieuDeNaissance) {
		this.lieuDeNaissance = lieuDeNaissance;
	}

	public LocalDate getDateAdhesion() {
		return dateAdhesion;
	}

	public void setDateAdhesion(LocalDate dateAdhesion) {
		this.dateAdhesion = dateAdhesion;
	}

	public String getNumeroVote() {
		return numeroVote;
	}

	public void setNumeroVote(String numeroVote) {
		this.numeroVote = numeroVote;
	}

	public String getPhoto() {
		return photo;
	}

	public void setPhoto(String photo) {
		this.photo = photo;
	}
    
	

	public String getTelephone() {
		return telephone;
	}

	public void setTelephone(String telephone) {
		this.telephone = telephone;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}
	
	public Set<Adresse> getAdresses() {
		return adresses;
	}
   @JsonSetter
	public void setAdresses(Set<Adresse> adresses) {
		this.adresses = adresses;
	}
	public Service getService() {
		return service;
	}
    @JsonSetter
	public void setService(Service service) {
		this.service = service;
	}
	@JsonIgnore
	public Set<Equipe> getEquipes() {
		return equipes;
	}
    @JsonSetter
	public void setEquipes(Set<Equipe> equipes) {
		this.equipes = equipes;
	}
	public Communaute getCommunaute() {
		return communaute;
	}
    @JsonSetter
	public void setCommunaute(Communaute communaute) {
		this.communaute = communaute;
	}
	public BureauVote getBureauVote() {
		return bureauVote;
	}
    @JsonSetter
	public void setBureauVote(BureauVote bureauVote) {
		this.bureauVote = bureauVote;
	}
	public Profession getProfession() {
		return profession;
	}
    @JsonSetter
	public void setProfession(Profession profession) {
		this.profession = profession;
	}

	public Long getId_adherent() {
		return id_adherent;
	}

	public String getMotPasse() {
		return motPasse;
	}

	public void setMotPasse(String motPasse) {
		this.motPasse = motPasse;
	}

	public String getSexe() {
		return sexe;
	}

	public void setSexe(String sexe) {
		this.sexe = sexe;
	}

	public Set<TypeAdherent> getTypeAdherents() {
		return typeAdherents;
	}

	public void setTypeAdherents(Set<TypeAdherent> typeAdherents) {
		this.typeAdherents = typeAdherents;
	}

	public Set<LigneFonction> getLignesFonction() {
		return lignesFonction;
	}

	public void setLignesFonction(Set<LigneFonction> lignesFonction) {
		this.lignesFonction = lignesFonction;
	}

	public String getRole() {
		return role;
	}

	public void setRole(String role) {
		this.role = role;
	}

	public void setId_adherent(Long id_adherent) {
		this.id_adherent = id_adherent;
	}
	
	
	
	
	
	
}

File: front-end/src/app/entities/adherent.ts
import { Serv } from './serv';
import { Adresse } from './adresse';
import { Communaute } from './communaute';
import { BureauVote } from './bureau-vote';
import { Profession } from './profession';
export class Adherent {

  private  _adresse:Adresse=new Adresse();
  private _communaute:Communaute;
  private _bureauVote:BureauVote;
  private _profession:Profession;
  private _service:Serv;
  constructor(private _id_adherent ?: Number,
        private _nom_adherent ?: string,
        private _prenom_adherent ?: string,
        private _cin ?:string,
        private _dateDeNaissance ?: Date,
        private _lieuDeNaissance ?: string,
        private _dateAdhesion?:Date,
        private _numeroVote ?: Number,
        private _photo ?:string,
        private _motPasse?:string,
        private _sexe ?:string,
        private _telephone ?:string,
        private _email ?: string,
        private _role?:string
        ) {
          this._adresse=new Adresse();
          this._communaute=new Communaute();
          this._bureauVote=new BureauVote();
          this._profession=new Profession();
          this._service=new Serv();
        }
        get id_adherent() {
            return this._id_adherent;
        }

        get nom_adherent() {
            return this._nom_adherent;
        }
        set nom_adherent(value: string) {
            this._nom_adherent = value;
        }
        get prenom_adherent() {
            return this._prenom_adherent;
        }
        set prenom_adherent(value: string) {
            this._prenom_adherent = value;
        }
        get cin(){
            return this._cin;
        }
        set cin(value:string){
            this._cin=value;
        }
        get dateDeNaissance(){
            return this._dateDeNaissance;
        }
        set dateDeNaissance(value:Date){
            this._dateDeNaissance=value;
        }
        get lieuDeNaissance(){
            return this._lieuDeNaissance;
        }
        set lieuDeNaissance(value:string){
            this._lieuDeNaissance=value;
        }
        get dateAdhesion(){
          return this._dateAdhesion;
        }
        set dateAdhesion(value:Date){
          this._dateAdhesion=value;
        }
        get numeroVote(){
            return this._numeroVote;
        }
        set numeroVote(value:Number){
            this._numeroVote=value;
        }
        get photo(){
          return this._photo;
        }
        set photo(value:string){
          this._photo=value;
        }
        get motPasse(){
          return this._motPasse;
        }
        set motPasse(value:string){
          this._motPasse=value;
        }
        get sexe(){
          return  this._sexe;
        }
        set sexe(value:string){
          this._sexe=value;
        }
        get telephone(){
          return this._telephone;
        }
        set telephone(value:string){
          this._telephone=value;
        }
        get email(){
          return this._email;
        }
        set email(value:string){
          this._email=value;
        }
        get adresse():Adresse{
          return this._adresse;
        }
        set adresse(value:Adresse){
          this._adresse=value;
        }
        get bureauVote():BureauVote {
          this._bureauVote=new BureauVote();
          return this._bureauVote;
        }
        set bureauVote(value:BureauVote){
          this._bureauVote=value;
        }
        get commnuate():Communaute{
          return this._communaute;
        }
        set communaute(value:Communaute){
          this._communaute=value;
        }
        get profession():Profession{
          return this._profession;
        }
        set profession(value:Profession){
          this._profession=value;
        }
        get service():Serv{
       return this._service;
        }
        set service(value:Serv){
          this._service=value;
        }
        get role():string{
          return this._role;
        }
        set role(value:string){
         this._role=value;
        }

}
